aws_embedded_metrics/serializers/log_serializer.py

```diff
@@ -53,24 +55,53 @@ def create_body() -> Dict[str, Any]:
         current_body: Dict[str, Any] = create_body()
         event_batches: List[str] = []
         num_metrics_in_current_body = 0
+        num_datapoints_in_current_body = 0
```

Comment:
I'm not sure I'm following why we need to track the number of datapoints in the body. There are 2 requirements:
1. You cannot have more than 100 metric names in an EMF event
2. A given metric name cannot have more than 100 samples

These two requirements bound the number of metric datapoints in an event to 10_000.

Reply:
I didn't understand the spec properly, I thought the limit of 100 samples was for the body. I will correct to limit to 100 samples per metric name.